### ISSUE TITLE
Mask not resized when using an Image input with tool="sketch" and a shape argument

### ISSUE THREAD

**Issue by leszekhanusz** [gradio-app/gradio#2054]
### Describe the bug

if we use an Image input with `tool="sketch"` and a `shape` argument, then only the image is resized to the shape and not the mask.

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Reproduction

```python
import gradio as gr                                                                                                     
                                                                                                                        
def run(sketch):                                                                                                        
    image = sketch["image"]                                                                                             
    mask = sketch["mask"]                                                                                               
                                                                                                                        
    return str(image.size), str(mask.size)                                                                              
                                                                                                                        
interface = gr.Interface(                                                                                               
    run,                                                                                                                
    inputs=[                                                                                                            
        gr.Image(interactive=True, type="pil", tool="sketch", shape=(512,512)),                                         
    ],                                                                                                                  
    outputs=[                                                                                                           
        gr.Textbox(label="Image shape"),                                                                                
        gr.Textbox(label="Mask shape"),                                                                                 
    ],                                                                                                                  
)                                                                                                                       
                                                                                                                        
interface.launch()
```

### Screenshot

_No response_

### Logs

```shell
No logs.
```


### System Info

```shell
Gradio v3.1.6
```


### Severity

annoying

**Comment by pngwn** [comment]
@leszekhanusz Can you see if this bug is still present in gradio 3.4, we have made a lot of changes to the sketch tool and i think this should be fixed now.

**Comment by leszekhanusz** [comment]
Nope, this is not fixed. I just tested with gradio 3.4 and the problem is still there.
Btw you can reproduce it very easily by:

- copy-pasting the reproduction code in a file
- run it
- go to http://127.0.0.1:7860
- upload any image with a different resolution than 512x512 px
- Click submit
- If it works then the mask shape should be (512, 512), like the image shape.

**Comment by pngwn** [comment]
Will take a look, thanks!

**Comment by pngwn** [comment]
We could also resize images (maintaining dimensions) on the server if images are uploaded that are too large. Depends on the reason for setting `max` really. If it is because of model performance then it might make sense but if the wrangling of large images itself is a bottleneck, then blocking the upload + simple validation on the server makes more sense. We could also resize in the browser.

**Comment by leszekhanusz** [comment]
As an example, here is how it is done for the [automatic1111 fork of Stable Diffusion](https://github.com/AUTOMATIC1111/stable-diffusion-webui):

* The shape argument is not used
* You can define the width and height of the generated picture
* there is a `resize_mode` input which defines how to fix the aspect ratio with the following options:
  * `Just resize` which resize the image and potentially change the aspect ratio
  * `Crop and resize` which will blindly crop the image in the middle to keep the aspect ratio
  * `Resize and fill` which will fill the picture on the sides with lines to keep the aspect ratio

An integrated cropping tool designed to keep a specific aspect ratio and giving the user the control and visibility about the potential cropping would definitely be a great addition to this library.

What would be absolutely needed in that case is that the shape of the image could be defined by other inputs (the width and height inputs in that case)

**Comment by abidlabs** [comment]
Just tested this on the latest version of gradio and it's fixed now:

<img width="606" alt="image" src="https://user-images.githubusercontent.com/1778297/220954678-7c33f5e1-3ddd-497d-8e9e-3e1f36e07940.png">

It was probably fixed as a result of PR #3277!

There's a separate issue (visible in the screenshot) of the image being resized incorrectly (but that is an existing issue here: #3093), so I'll go ahead and close this issue. I'll also move your suggestions about cropping to this issue here: #1451
